controllers/edgedeployment_controller.go

```diff
@@ -285,3 +358,25 @@ func mergeErrorMessages(errs []error) string {
 	}
 	return message
 }
+
+func SplitEdgeDevices(s []managementv1alpha1.EdgeDevice, x int) [][]managementv1alpha1.EdgeDevice {
+	if x <= 0 {
+		return nil
```

Comment:
That should be an error.

Reply:
It's only in theory. It is an internal function to be used in this file. I just did that so tests and such won't fail.
I don't want to make this function return error and change all calls to it just so in theory, maybe, someday, someone might do something terribly wrong. I think it is better to let Go panic in such a case.
I will change it to uint and check only x==0.

Reply:
But will it panic? From what I see it shouldn't; at line https://github.com/jakub-dzon/k4e-operator/pull/38/files#diff-0d2d8c9ff48a926eddf2ba67be4f7dc0635e7b88376946057eba6dabfb0931ecR280 it will return nil immediately, which will return nil-error array, which would be handled gracefully. Nobody will notice the misconfiguration.
Maybe it should be validated in main and cause the app to crash right away?

Reply:
it won't panic because you asked for a change. I would have left it as it was originally.
I think there's no need to handle every theoretical problem. I can not imagine someone setting the value to 0 on purpose.
and even someone did, the effect would be immediate (e.g. no deployments start/stop in devices).
But again, for the sake of saving time debating I will add validation in main.